### PR TITLE
Deleting unnecessary 'export' from helper functions in dialog capability

### PR DESCRIPTION
**Pull request by lakhveerk**
For more information about how to contribute to this repo, visit [this page](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md).

## Description

> Currently we are exporting some of the functions just to unit test them which is not a good practice. This PR is deleting un-necessary 'export' and fixing the unit tests accordingly. 

> https://office.visualstudio.com/MetaOS/_workitems/edit/6670382

### Main changes in the PR:

1. Deleted 'export' from helper function in dialog file
2. Deleted unused and unnecessary functions in dialog file
3. Fixed Unit tests

### Validation performed:

1. <Step 1>
2. <Step 2>

### Unit Tests added:

> Unit tests are required for all changes. If no unit tests were added as part of this change, please explain why they aren't necessary.

<Yes>

### End-to-end tests added:

<Not applicable>

### Change file added:

> Ensure the change file meets the [formatting requirements](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md#change-log-using-beachball).

<Yes>

### CODE REVIEW

File: packages/teams-js/src/public/dialog.ts [hunk 1]
```diff
@@ -514,53 +512,12 @@ export namespace dialog {
...
       return dialogInfo;
     }
-
-    /**
-     * @hidden
-     * Converts {@link TaskInfo} to {@link AdaptiveCardDialogInfo}
-     * @param taskInfo - TaskInfo object to convert
-     * @returns - converted AdaptiveCardDialogInfo
-     */
-    export function getAdaptiveCardDialogInfoFromTaskInfo(taskInfo: TaskInfo): AdaptiveCardDialogInfo {
```
Comment (lakhveerk) [inline packages/teams-js/src/public/dialog.ts:531]: These functions were hanging out for no reason. They are not used anywhere.

File: packages/teams-js/src/public/tasks.ts [hunk 2]
```diff
@@ -126,12 +126,18 @@ export namespace tasks {
   }
 
   /**
+   * @hidden
```
Comment (lakhveerk) [inline packages/teams-js/src/public/tasks.ts:129]: I am not sure why functions like this and `getBotUrlDialogInfoFromTaskInfo` and `getUrlDialogInfoFromTaskInfo` are not marked internal and hidden. We do not expect users to call them directly. These are just the helper functions. 
Is it a major change to mark them hidden and internal now?

**Review by jadahiya-MSFT — APPROVED** [review]

**Review by shrshindeMSFT — APPROVED** [review]
LGTM! :speedboat: speed

**Review by TrevorJoelHarris — CHANGES_REQUESTED** [review]

File: change/@microsoft-teams-js-a26c1e08-e775-4b10-9284-03bd7bc8c64c.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
```
[the remark this answers is not in the document]
Comment (lakhveerk) [reply change/@microsoft-teams-js-a26c1e08-e775-4b10-9284-03bd7bc8c64c.json]: That makes sense. Changed it to minor :)

File: packages/teams-js/src/public/tasks.ts (hunk 2 above)
Comment (lakhveerk) [reply packages/teams-js/src/public/tasks.ts]: sounds good! Another point is it is already deprecated. So, there is no need to release a major change for this.

**Review by TrevorJoelHarris — APPROVED** [review]

**Review by shrshindeMSFT — APPROVED** [review]

**Pull request merged**